legend-sdlc-entity-maven-plugin/src/main/java/org/finos/legend/sdlc/entities/EntityReserializer.java

```diff
@@ -114,7 +115,13 @@ private List<String> reserializeFile(Path sourceFile, Path targetDirectory, bool
         {
             if (enforceOneEntityPerFile)
             {
-                entities = Collections.singletonList(this.sourceSerializer.deserialize(inputStream));
+                Entity singleEntity = this.sourceSerializer.deserialize(inputStream);
+                Path expectedPath = sourceSerializer.filePathForEntity(singleEntity, Paths.get(""));
+                if (!sourceFile.endsWith(expectedPath))
```

Comment:
We should pass sourceDirectory into this method. Then we can do the check precisely. That will also make the exception message more precise.
```suggestion
                Path expectedPath = sourceSerializer.filePathForEntity(singleEntity, sourceDirectory);
                if (!sourceFile.equals(expectedPath))
```